### PR TITLE
feat: initial backend implementation for custom realm request

### PR DESCRIPTION
**Pull request by NithinKuruba**
Fixes [SSOTEAM-1095](https://bcdevex.atlassian.net/browse/SSOTEAM-1095)

### CODE REVIEW

File: app/package.json [hunk 1]
```diff
@@ -42,22 +50,34 @@
...
     "@types/pg-format": "^1.0.2",
     "@types/react": "^18.0.10",
     "@types/validator": "^13.7.4",
     "@types/xml2js": "^0.4.11",
+    "@types/yup": "^0.32.0",
     "eslint": "^8.16.0",
     "eslint-config-next": "12.1.6",
+    "prettier": "^3.0.3",
+    "prettier-plugin-prisma": "^5.0.0",
+    "prisma": "^5.5.2",
```
Comment (jlangy) [inline app/package.json:80]: Should prisma be a regular dependency? Or does it only need @prisma/client

File: app/pages/api/realms/[id].ts [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+              realmId: parseInt(req.query.id as string, 10),
+              eventCode: EventEnum.REQUEST_REJECT_SUCCESS,
+              idirUserId: username,
+              details: req.body,
+            });
+            updateRequest.approved = false;
+          }
+        }
+
+        updatedRealm = await prisma.roster.update({
```
Comment (jlangy) [inline app/pages/api/realms/[id].ts:151]: Does the put request validate ownership? Not sure if its in that `getUpdateRealmSchemaByRole` funtion

**Review by jlangy — COMMENTED** [review]
Just left two questions

File: app/pages/api/realms/[id].ts (hunk 2 above)
Comment (NithinKuruba) [reply app/pages/api/realms/[id].ts:151]: Only logged in users would only be able to view their owned realms but I have updated the where condition to be more descriptive.

File: app/package.json (hunk 1 above)
Comment (NithinKuruba) [reply app/package.json:81]: prisma is just a cli

**Review by jlangy — APPROVED** [review]

**Pull request merged**